Userland/Libraries/LibSyntax/Language.cpp

```diff
@@ -72,4 +72,70 @@ StringView common_language_extension(Language language)
     VERIFY_NOT_REACHED();
 }
 
+Optional<Language> language_from_name(StringView name)
+{
+    if (name.equals_ignoring_case("CMake"sv))
+        return Language::CMake;
+    if (name.equals_ignoring_case("CMakeCache"sv))
+        return Language::CMakeCache;
+    if (name.equals_ignoring_case("Cpp"sv))
+        return Language::Cpp;
+    if (name.equals_ignoring_case("CSS"sv))
+        return Language::CSS;
+    if (name.equals_ignoring_case("GitCommit"sv))
+        return Language::GitCommit;
+    if (name.equals_ignoring_case("GML"sv))
+        return Language::GML;
+    if (name.equals_ignoring_case("HTML"sv))
+        return Language::HTML;
+    if (name.equals_ignoring_case("INI"sv))
+        return Language::INI;
+    if (name.equals_ignoring_case("JavaScript"sv))
+        return Language::JavaScript;
+    if (name.equals_ignoring_case("PlainText"sv))
+        return Language::PlainText;
+    if (name.equals_ignoring_case("SQL"sv))
+        return Language::SQL;
+    if (name.equals_ignoring_case("Shell"sv))
+        return Language::Shell;
+
+    return {};
+}
+
+Optional<Language> language_from_filename(LexicalPath const& file)
+{
+    if (file.title() == "COMMIT_EDITMSG"sv)
+        return Language::GitCommit;
+
+    auto extension = file.extension();
+    VERIFY(!extension.starts_with('.'));
+    if (extension == "cmake"sv || (extension == "txt"sv && file.title() == "CMakeLists"sv))
+        return Language::CMake;
+    if (extension == "txt"sv && file.title() == "CMakeCache"sv)
+        return Language::CMakeCache;
+    if (extension == "c"sv || extension == "cc"sv || extension == "cxx"sv || extension == "cpp"sv || extension == "c++"
+        || extension == "h"sv || extension == "hh"sv || extension == "hxx"sv || extension == "hpp"sv || extension == "h++"sv)
```

Comment:
I know you copied this from HS, but let me introduce you to my friend while you're here:
```c++
extension.is_one_of("c"sv, "cc"sv, "cxx"sv, "cpp"sv, "c++"sv, "h"sv, "hh"sv, "hxx"sv, "hpp"sv, "h++"sv)
```

Reply:
Nice!